dashboard/static/js/grade_distribution.js

```diff
@@ -37,9 +38,10 @@ var makeGraph = function () {
         height = height - margin.top - margin.bottom;
         g = svg.append("g")
             .attr('transform', 'translate(' + margin.left + ',' + margin.top + ')');
+        var domain_values = (data_more_than_100_percent) ? [0, 110] : [0, 100]
```

Comment:
Is 110 big enough to accommodate extra points?

Reply:
Yeah, maybe it would be nice to see what the visualization looks like in this case. 

Extra credit in grading has somewhat confused me. Either there's still a hard limit of 100% and anything over it doesn't get you anything extra, or grades over 100% give you something different. It looks like in Canvas, the grading schema can't go over 100% but students can still score over 100%. https://community.canvaslms.com/thread/2127

I wonder if our visualization needs to account for this? Maybe have a little "piece" on the end that's "up-to" 10% in width with a tooltip that says how far over 100% the students total grade is? Is this is what this code is attempting to do? Maybe an idea for phase 2.

Reply:
Should be good or have to do the math based on the percent appropriate. I could also tell D3 that fix the domain limit based on values in the list. The problem with that is if the class is in 90-100 the distribution range it looks odd.  I will lie if I say i did not think is 110 enough? But being practical will the % grade ever grow higher than  110%, I felt unlikely so I made the hardcoded number.

Reply:
![screen shot 2018-11-02 at 9 30 30 am](https://user-images.githubusercontent.com/8579775/47918162-1557f200-de82-11e8-8504-7626f0e4c698.png)
![screen shot 2018-11-02 at 9 30 46 am](https://user-images.githubusercontent.com/8579775/47918165-1852e280-de82-11e8-9ff3-6313bcf90e5c.png)

Reply:
Yeah this looks good. This 110% only shows if there are grades over 100% right? I think it's nice to show. If there are any grades over 110% it could probably change the label to "110%+" and just show them on the "110%+ line". We might need to add this logic to the files page too. 

I think we *could* have just even left it at 100% and put all the grades >= 100% on a "100%+" line but it's probably nice to have this extra bit of visualization.

Reply:
Just wait till we have the gamification class that goes up to 1000% or something someday. ;)

Reply:
OK, I have made the upper limit adaptive based on the data. so if we have course with grade distribution like 150% the view will serve it than abandoning scores as before
@jonespm I like your idea of making it something like 110%+, I think code needs to change the logic around `MyGrade` line if a student has a score 115% (say) that means you i have put the orange line on 110%+ marker as that is the last marker on the line. So I feel the docs needs to be updated with this feature. We will keep it simple for now for everybody. We can think about this idea for next phase.

Reply:
I agree with the above comments. I think a 110%+ column is the best visualization option. We push the distribution of the graph to far to the left in scenarios where a handful of students score in 1.4, 1.5, and 10.1 percentages.